Remove mdc attributes prefix

SHaaD94 opened the pull request:
As discussed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9506 removing MDC attributes prefix for both log4j and logback.

laurit commented on instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java:
```diff
@@ -159,23 +157,19 @@ void captureMdcAttributes(AttributesBuilder attributes, Map<String, String> mdcP
     }
   }
 
-  public static AttributeKey<String> getMdcAttributeKey(String key) {
```
why did you decide to remove the attribute cache?

SHaaD94 replied:
I assumed it was required only because of prefix addition, but now since we don't do it anymore it shouldn't be required, since the rest places there attributes are created aren't using the cache.

laurit replied:
It was used to reduce creation of `AttributeKey` instances

SHaaD94 replied:
Alright, I'll rollback the change

SHaaD94 replied:
rolled back

laurit reviewed (commented):
Prefix is also added in
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/jboss-logmanager/jboss-logmanager-appender-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/appender/v1_1/LoggingEventMapper.java#L131
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L179
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L184

SHaaD94 commented:
> Prefix is also added in
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/jboss-logmanager/jboss-logmanager-appender-1.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jbosslogmanager/appender/v1_1/LoggingEventMapper.java#L131
> 
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L179
> 
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c7617dc06800c15ef5fbe729688c3fcbe2c1c0d4/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L184

removed these as well

trask commented:
@laurit @mateuszrzeszutek do you think we should hold this for 2.0?

jack-berg commented:
This is related to: https://github.com/open-telemetry/opentelemetry-java/pull/5880

If we put structured key value pairs in the body as AnyValue, there is no chance of collision with the attributes and the prefix is uneeded.

trask commented:
one concern I have with putting MDC attributes in the body is that they won't be available to attribute-based processors (e.g. in the collector), or searchable in backends in the same way other log attributes are, at least without additional backend work / interpretation of the log bodies

laurit commented:
> @laurit @mateuszrzeszutek do you think we should hold this for 2.0?

I think it is fine either way. We could put it in 2.0 along with other breaking changes or just outline this as a breaking change in the release notes of the next release.

AlchemyDing commented:
Do we need a configuration to enable the prefix of MDC?

trask commented:
hi @SHaaD94! we are preparing to release v2.0 soon, can you rebase your PR and check any CI failures? thx

SHaaD94 commented:
> hi @SHaaD94! we are preparing to release v2.0 soon, can you rebase your PR and check any CI failures? thx

Hello @trask. Done

jack-berg commented:
> one concern I have with putting MDC attributes in the body is that they won't be available to attribute-based processors (e.g. in the collector), or searchable in backends in the same way other log attributes are, at least without additional backend work / interpretation of the log bodies

@trask sorry for the delay replying. I don't think we should put MDC attributes in the body - I think we should [MapMessage](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L141C1-L153C6) / [KeyValuePair](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java#L219-L229) in the body. 

Do you think we should do this and if so, do you think its worthwhile to bundle in with 2.0?

mateuszrzeszutek commented:
> @trask sorry for the delay replying. I don't think we should put MDC attributes in the body - I think we should [MapMessage](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java#L141C1-L153C6) / [KeyValuePair](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java#L219-L229) in the body.

Sorry for even more delayed reply 😂 
I totally agree with that -- I think MDC attributes are more like metadata of the log message, not the actual data of it; and log record attributes seem to be a better place for them than body.

> Do you think we should do this and if so, do you think its worthwhile to bundle in with 2.0?

Translating the `MapMessage` to log record body? Is it strictly necessary for 2.0, or can we introduce that a bit later?

trask commented:
Discussed in SIG. I will push commit to revert the map_message portion in order to avoid churn since we are planning to move map messages to log body in the future.
